microsoft/fhir-server
# Return BadRequestException with valid message when input json body is invalid

## Description
For invalid json body requests Fhir server returns 500. If the input json is not in the correct format then we are parsing the body at [FhirJsonInputFormatter](https://github.com/microsoft/fhir-server/blob/main/src/Microsoft.Health.Fhir.Shared.Api/Features/Formatters/FhirJsonInputFormatter.cs) and passing the initial validations for required fields at [ModelAttributeValidator](https://github.com/microsoft/fhir-server/blob/main/src/Microsoft.Health.Fhir.Core/Features/Validation/IModelAttributeValidator.cs).

 e.g. 
If the body contains _Coverage.status = ""_, then after parsing _Coverage.status = null_ & _Coverage.statusElement = null_, resulting into minimum cardinality error as expected.
If the body contains _Coverage.status =_ , then after parsing _Coverage.status = null_ & _Coverage.statusElement = {value=null}_, which passes the Firely validation and [CodeToTokenSearchValueConverter](https://github.com/microsoft/fhir-server/blob/main/src/Microsoft.Health.Fhir.Core/Features/Search/Converters/CodeToTokenSearchValueConverter.cs) returns null causing search index entry to fail.

 In this PR we are returning BadRequestException with a valid message instead of 500

## Related issues
Addresses [85266](https://microsofthealth.visualstudio.com/Health/_workitems/edit/85266)

## Testing
1. Check POST Coverage request with - "status" : ,
2. Check POST Member-Match request with - "status" : ,
3. Check POST Claim request with - "use": , 
4. Unit tests

## FHIR Team Checklist
- Review [squash-merge requirements](https://github.com/microsoft/fhir-server/blob/master/SquashMergeRequirements.md)

### Semver Change ([docs](https://github.com/microsoft/fhir-server/blob/master/docs/Versioning.md))
Patch|Skip|Feature|Breaking (reason)
